src/labileFilter.js

```diff
@@ -1,238 +1,179 @@
+  // @TODO: import function
+  let fitArray = [];
+  for (let i = 0; i < 100; i++) {
+    fitArray[i] = simulatedAnnealing(fitOptions);
+  }
+  let optimumArray = fitArray.map((x) => x.optimum);
+  let optimumSD = standardDeviation(optimumArray);
+
+  //@TODO: use ml-array-min
+  let optimum = optimumArray.reduce(function (a, b) {
+    return Math.min(a, b);
+  });
+
+  //@TODO: check if index exist
+  let fit = fitArray[optimumArray.indexOf(optimum)];
+  if (fit.optimum > 3) {
+    labileProtonRange[0] = {
+      info: {
+        from: maxWidthPeakData.x - maxWidthPeakData.width,
+        to: maxWidthPeakData.x + maxWidthPeakData.width,
+      },
+    };
+  }
+  let fitData = {
+    x: subSpectra.x,
+    y: gaussian(fit.best, subSpectra.x),
+  };
+
+  let realWidth = Math.abs(
+    subXSpectra[subXSpectra.length - 1] - subXSpectra[0],
+  );
+  let bounds = {
+    x: [subXSpectra[0], subXSpectra[subXSpectra.length - 1]],
+    y: [0.5, 0.5],
+  };
+
+  let maxIntensity = subSpectra.y.reduce(function (a, b) {
     return Math.max(a, b);
-    });
-    let initialParameters = initialDataGenerator(subSpectra.x, subSpectra.y)
-    let boundsPlot = {'data' : []
-    }
-    let lowerBoundData = gaussian(initialParameters.lowerBound,subSpectra.x )
-    let upperBoundData = gaussian(initialParameters.upperBound,subSpectra.x)
-    let guessData = gaussian(initialParameters.guess,subSpectra.x)
-    let boundsData = [lowerBoundData, upperBoundData, guessData];
-    for(let i = 0; i <  boundsData.length; i++){
-        boundsPlot.data[i] = {
-        'x' : (subSpectra.x),
-        'y' : (boundsData[i]),
-        'style' : {"lineColor":"rgb(230,0,0)","lineWidth":1,"lineStyle":1}
-        }
-    }
-    let errorFunction = getErrorFunction(gaussian, subSpectra);
-    let fitOptions = {
-        initialT: 1.0,
-        objetiveFunction: errorFunction,
-        minT: 1e-10,
-        minFunctionValue: -1e+10,
-        maxRejection: 2500, 
-        maxRun: 2000,
-        maxAccept: 15,
-        k: 1,
-        alpha: 0.95,
-        energyNorm: 1e-8,
-        guess: initialParameters.guess,
-        lowerBound: initialParameters.lowerBound,
-        upperBound: initialParameters.upperBound
-    }
-    
-    // let subXMin =  x.reduce(function(a, b) {
-    // return Math.min(a, b);
-    // });
-    let fitArray = []
-    for (let i = 0; i < 100; i++){
-        fitArray[i] = simulatedAnnealing(fitOptions) 
-    }
-    let optimumArray = fitArray.map( x => x. optimum)
-    API.createData('optimumArray', optimumArray);
-    let optimumSD = standardDeviation(optimumArray)
-    console.log(optimumSD, 'optimumSD')
-    let optimum = optimumArray.reduce(function(a, b) {
+  });
+  //@TODO: check this function
+  let boundsTest = gaussianBounds(subSpectra.x, subSpectra.y);
+  let maxIntensityIndex = subSpectra.y.indexOf(maxIntensity);
+  let subXMax = subSpectra.x.reduce(function (a, b) {
+    return Math.max(a, b);
+  });
+  let subXMin = subSpectra.x.reduce(function (a, b) {
```

Comment:
ml-min-array

Reply:
los bucles for son mucho mas rapidos que un reduce porque este ultimo tiene que garantizar la function de entrada